app/public/src/game/game-container.ts

```diff
@@ -70,7 +70,7 @@ class GameContainer {
   initializeSimulation(simulation: Simulation) {
     if (
       simulation.bluePlayerId === this.player?.id ||
-      simulation.redPlayerId === this.player?.id
+      (simulation.redPlayerId === this.player?.id && !simulation.isGhostBattle)
```

Comment:
this is the actual bug, we were displaying the wrong simulation client-side because we now preserve player.id when attributing the ghost matchup.